mplc/partner.py

```diff
@@ -123,6 +124,36 @@ def shuffle_labels(self, proportion_shuffled):
         for i in idx:
             np.random.shuffle(self.y_train[i])
 
+    def duplicate_data(self, partner_copied, proportion):
+        if not 0 <= proportion <= 1:
+            raise ValueError(
+                f"The proportion of labels to corrupted was {proportion} but it must be between 0 and 1."
+            )
+        # Select the indices where the label will be off-set
+        n = int(len(self.y_train) * proportion)
+
+        if len(partner_copied.x_train) > n:
+            idx = sample(list(range(len(self.y_train))), n)
+        else:
+            n = int(len(partner_copied.y_train))
+            idx = sample(list(range(len(self.y_train))), n)
+            logger.warning(f"The partner which dataset would have been copied does not have enough data"
+                           f" to corrupt {proportion * 100}% of this partner's dataset."
+                           f" Only {np.round((n / len(self.y_train)), 2) * 100} percent will be corrupted")
+        self.y_train[idx] = partner_copied.y_train[:len(idx)]
+        self.x_train[idx] = partner_copied.x_train[:len(idx)]
```

Comment:
Hmmm, isn't there a risk there, not to deepcopy what we want?
`partner_copied.y_train[:len(idx)]` is mutable, `self.y_train[idx]` is only a pointer to it (_I think but not sure though?_), so if it changes later on for some reason... Just wondering

Reply:
I tested that, and I haven't had any problem. It's not list but slices of a numpy array. I think it work because I use idx.